### ISSUE TITLE
What is the Default implementation in the macro used for

### ISSUE THREAD

**Issue by dovreshef** [dalance/structopt-toml#8]
Hi

From the  `impl_structopt_for_struct` function the following Default trait implementation is generated:
```rust
        impl Default for #name {
            fn default() -> Self {
                let args = vec!["bin"];
                #name::from_iter(args.iter())
            }
        }
```
It is then called by serde if there is no value in the toml string.
But I don't understand why its implementation is the way it is.

It fails me in the following case:

```rust
use structopt::StructOpt;
use structopt_toml::StructOptToml;
use serde_derive::Deserialize;


#[derive(Debug, Deserialize, StructOpt, StructOptToml)]
#[serde(default)]
struct Cmdline {
    #[structopt(long = "url", required_unless = "path")]
    url: Option<String>,

    #[structopt(long = "path", conflicts_with = "url")]
    path: Option<String>,
}

fn main() {
    let data = String::new();

    // let opt = Cmdline::from_args();
    let opt = Cmdline::from_args_with_toml(&data).unwrap();
    println!("{:?}", opt);
}
```

Trying to run the example above while supplying any one of the two arguments will fail:

```shell
19:01 $ cargo run --  --url https://example.com
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/so --url 'https://example.com'`
error: The following required arguments were not provided:
    --url <url>

USAGE:
    bin [OPTIONS] --url <url>

For more information try --help
```

From the `bin` for the binary name I can see that's it's taken from the Default implementation. But I'm not clear to what purpose..

**Comment by dalance** [comment]
This `Default` implementation is used to get the default value defined by `structopt` like below:

```
struct Cmdline {
    #[structopt(default_value = "100")]
    x: usize,
}
```

Surely this approach will be failed by `required*` option.
To avoid it, I think the `Default` implementation should parse `default_value` to get the default value.

**Comment by dovreshef** [comment]
I think, that would still break cases like above, since it will not take care of all the `required_unless`, `required_if`, `requires`.

What about replacing it with
```rust
impl Default for #name {
    fn default() -> Self {
        #name::from_args()
    }
}
```

This way, clap will still enforce all its variants, and if they fail - will fail the process as today.

The benefit I think would be:
1. It would use the args supplied by the user, including default_values supplied to clap, but only as a fall back, so the order would be:
* Arguments by command line
* Arguments in toml file
* Arguments and default values by command line

2. It would use the correct name of the binary in the error it will output, in cases where there are missing required values.

What do you think?

**Comment by dalance** [comment]
I merged #9, and released v0.4.5.
Thank you for your contribution.

**Comment by dovreshef** [comment]
Thanks!!
